uplink/src/collector/downloader.rs

```diff
@@ -308,46 +300,46 @@ impl FileDownloader {
         Ok((file, file_path))
     }
 
+    // Retry mechanism tries atleast 3 times before returning a download error
+    async fn retry_thrice(&mut self, url: &str, mut download: DownloadedFile) -> Result<(), Error> {
```

Comment:
We don't need retry_thrice. We can indefinitely retry (probably with 1s poll) until timeout

Reply:
We can do that in a separate PR I feel like